libraries/AP_IRLock/pixy_parser.cpp

```diff
@@ -0,0 +1,272 @@
+    // printf("Pixy_buf: ");
+//    for (size_t i=0; i<PIXY_PARSER_PIXY_BUF_SIZE; i++) {
+        // printf("%u, ", pixy_buf[i]);
+//    }
+    // printf("\n");
+
+
+/*
+    if (writebuf.count > 1) {
+        if (!AP_HAL::millis() <= multiple_count_start_timestamp + 500) {
+            multiple_count_start_timestamp = AP_HAL::millis(); 
+        }
+    }
+
+*/
+
+
+    if (validity == MESSAGE_VALID_SOF) {
+        if (!(writebuf.count >= PIXY_PARSER_MAX_BLOBS)) {
+            pixy_blob received_blob;
+            received_blob.center_x = pixy_buf[8] | (uint16_t)pixy_buf[9]<<8;    //Extract blob information
+            received_blob.center_y = pixy_buf[10] | (uint16_t)pixy_buf[11]<<8;
+            received_blob.width = pixy_buf[12] | (uint16_t)pixy_buf[13]<<8;
+            received_blob.height = pixy_buf[14] | (uint16_t)pixy_buf[15]<<8;
+
+            if (writebuf.count != 0) {
+                swap_buffer();  //swap buffer
+//                if_swap_buffer = 1;
+                if_swap = 1;
+
+                writebuf.count = 0;     //Turn the count to 0
+                // printf("@@@SEE THIS----   [%u, %u, %u, %u], ", (unsigned)received_blob.center_x, (unsigned)received_blob.center_y, (unsigned)received_blob.width, (unsigned)received_blob.height);
+
+                write_buffer(received_blob);  //Writing inside the buffer(the count increment happens inside the write_buffer())
+//                return true;
+            }
+            else {
+                write_buffer(received_blob);    //Writing inside the buffer(the count increment happens inside the write_buffer())
+            }
+        }
+        empty_pixyBuf();
+    }
+
+    if (validity == MESSAGE_VALID_BLOCK) {     ///-----------------------------how can I store the block inside a frame?-----------------
+        if (!(writebuf.count >= PIXY_PARSER_MAX_BLOBS)) {
+            pixy_blob received_blob;
+            received_blob.center_x = pixy_buf[6] | (uint16_t)pixy_buf[7]<<8; //Extract blob information
+            received_blob.center_y = pixy_buf[8] | (uint16_t)pixy_buf[9]<<8;
+            received_blob.width = pixy_buf[10] | (uint16_t)pixy_buf[11]<<8;
+            received_blob.height = pixy_buf[12] | (uint16_t)pixy_buf[13]<<8;
+            // printf("@@@SEE THIS----   [%u, %u, %u, %u], ", (unsigned)received_blob.center_x, (unsigned)received_blob.center_y, (unsigned)received_blob.width, (unsigned)received_blob.height);
+
+            write_buffer(received_blob);  //Writing inside the buffer
+        }
+        empty_pixyBuf();    //Emptying The pixy_buf once read the SOF into 'writebuf'       
+    }
+
+    if (validity == MESSAGE_INVALID) {  // If message invalid, wait and read 2 bytes
+        pixy_len--;     ///----------------------------------------QUES----------------------------------------------
+        memmove(pixy_buf, pixy_buf+1, pixy_len);    // This discards the first block of the pixy_buf and makes pixy_buf have everything except the fisrt block value
```

Comment:
i don't understand how this works. Looks like a byte array, so this shifts by one byte, but comment implies its a block?